src/expound/paths.cljc

```diff
@@ -202,10 +202,14 @@
     1
 
     (and (vector? x) (vector? y))
-    (first (filter #(not= 0 %) (map compare-path-segment x y)))
+    (->> (map compare-path-segment x y)
+         (remove #{0})
+         first)
 
     :else
     (compare x y)))
 
 (defn compare-paths [path1 path2]
-  (first (filter #(not= 0 %) (map compare-path-segment path1 path2))))
+  (->> (map compare-path-segment path1 path2)
+       (remove #{0})
```

Comment:
This chunk of code is identical to the one above. Perhaps `compare-path-segment` and  `compare-paths` could be mutually recursive?

Didn't want to try it out, not super familiar with the code yet.

Reply:
Yeah, good idea!